bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -249,15 +318,11 @@ private void updateGuiding(final TargetEnvironment env) {
 
         // Update the guiding feedback.
         final ISPObsComponent node = getContextTargetObsComp();
-        final SPTarget      target = TargetSelection.get(env, node);
+        final SPTarget      target = TargetSelection.getTargetForNode(env, node).getOrNull();
         _w.detailEditor.gfe().edit(ctx, target, node);
     }
 
 
-    private static boolean enablePrimary(final SPTarget target, final TargetEnvironment env) {
-        return env.getBase() != target && !env.getUserTargets().contains(target);
-    }
```

Comment:
I folded the behaviour into `UpdateUIForTarget` and `UpdateUIForGroup`, since these are now handled differently with respect to the primary, and that was the only place they were used.